app/models/document.rb

```diff
@@ -62,7 +62,16 @@ def self.param_value(params, key)
         params.fetch(:"#{key.to_s}(2i)"),
         params.fetch(:"#{key.to_s}(3i)")
       ]
-      format_date.delete_if(&:empty?).join("-")
+      format_date.delete_if(&:empty?)
```

Comment:
I think it would be good to move some of this increasingly Date-centric logic into DateHelper _somehow_